docs/site/express-with-lb4-rest-tutorial.md

```diff
@@ -0,0 +1,358 @@
+? window.localStorage key to use for persistence (browser only):
+? Full path to file for persistence (server only): ./data/ds.json
+
+  create src/datasources/ds.datasource.json
+  create src/datasources/ds.datasource.ts
+  update src/datasources/index.ts
+
+Datasource ds was created in src/datasources/
+```
+
+Similar to the `Todo` example, let's create the `ds.json` by creating a data
+folder at the application's root.
+
+```sh
+$ mkdir data
+$ touch data/ds.json
+```
+
+Then copy and paste the following into the `ds.json` file:
+
+```json
+{
+  "ids": {
+    "Note": 3
+  },
+  "models": {
+    "Note": {
+      "1": "{\"title\":\"Things I need to buy\",\"content\":\"milk, cereal, and waffles\",\"id\":1}",
+      "2": "{\"title\":\"Great frameworks\",\"content\":\"LoopBack is a great framework\",\"id\":2}"
+    }
+  }
+}
+```
+
+#### Add Note Repository
+
+To create the repository, run the `lb4 repository` command and choose the
+`DsDataSource` and `Note` model, as follows:
+
+```sh
+$ lb4 repository
+? Please select the datasource DsDatasource
+? Select the model(s) you want to generate a repository Note
+   create src/repositories/note.repository.ts
+   update src/repositories/index.ts
+
+Repository Note was created in src/repositories/
+```
+
+#### Add Note Controller
+
+To complete the `Note` application, create a controller using the
+`lb4 controller` command and the following prompts:
+
+```sh
+$ lb4 controller
+? Controller class name: note
+? What kind of controller would you like to generate? REST Controller with CRUD functions
+? What is the name of the model to use with this CRUD repository? Note
+? What is the name of your CRUD repository? NoteRepository
+? What is the type of your ID? number
+? What is the base HTTP path name of the CRUD operations? /notes
+   create src/controllers/note.controller.ts
+   update src/controllers/index.ts
+
+Controller note was created in src/controllers/
+```
+
+### Express Class
+
+Create a new file **src/server.ts** to create your Express class:
+
+```sh
+npm install --save express
+npm install --save-dev @types/express
+```
+
+{% include code-caption.html content="src/server.ts" %}
+
+```ts
+import * as express from 'express';
+
+export class ExpressApplication {
+  constructor() {}
+}
+```
+
+Create two properties, the Express application instance and LoopBack application
+instance:
+
+```ts
+import {NoteApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+import * as express from 'express';
+
+export class ExpressApplication {
+  private app: express.Application;
+  private lbApp: NoteApplication;
+
+  constructor(options: ApplicationConfig = {}) {
+    this.app = express();
+    this.lbApp = new NoteApplication(options);
+  }
+}
+```
+
+Now, inside the constructor, we're going to add the basepath and expose the
+front-end assets via Express:
+
+```ts
+this.lbApp.basePath('/api');
+this.app.use(this.lbApp.requestHandler);
+```
```

Comment:
Now we should be able to use `this.app('/api', this.lbApp.requestHandler)`. The bug was fixed by https://github.com/strongloop/loopback-next/pull/2387

Reply:
@raymondfeng I updated it to use `this.app('/api', this.lbApp.requestHandler)`, but the `/api/explorer` endpoint still leads to `Fetch error Not Found /openapi.json`. But it works if I go to: http://explorer.loopback.io/?url=http://127.0.0.1:3000/api/openapi.json.